File: app/src/main/java/com/example/requestapp/ui/login/LoginActivityContract.java
package com.example.requestapp.ui.login;

import android.content.Context;

public interface LoginActivityContract {

    interface Presenter {

        void onLoginClicked(Context context, String nickName, String password);

        boolean isLoggedIn();

    }

    interface View {

        void onLoginSuccessful(String message);

        void onLoginFailure(String message);

        void onProcessStart();

        void onProcessEnd();

    }
}

File: app/src/main/java/com/example/requestapp/ui/login/LoginActivityContract.kt
package com.example.requestapp.ui.login

import android.content.Context

interface LoginActivityContract {
    interface Presenter {
        fun onLoginClicked(context: Context?, nickName: String?, password: String?)
        fun isLoggedIn(): Boolean
    }

    interface View {
        fun onLoginSuccessful(message: String?)
        fun onLoginFailure(message: String?)
        fun onProcessStart()
        fun onProcessEnd()
    }
}